src/main/java/gregtech/common/terminal/app/game/maze/MazeApp.java

```diff
@@ -0,0 +1,238 @@
+    private int timer = 0;
+    private int mazesSolved = 0;
+    private float speed = 25;
+    private int lastPlayerInput = -2;
+    public static int MAZE_SIZE = 9;
+    private List<Integer> movementStore = new ArrayList<>();
+    private boolean lastPausePress;
+
+    public MazeApp() {
+        super("maze");
+    }
+
+    public AbstractApplication createAppInstance(TerminalOSWidget os, boolean isClient, NBTTagCompound nbt) {
+        MazeApp app = new MazeApp();
+        if (isClient) {
+            app.setOs(os);
+            app.addWidget(new ImageWidget(5, 5, 333 - 10, 232 - 10, TerminalTheme.COLOR_B_2));
+            // Gamestate 0: Title
+            app.addWidget(new LabelWidget(333 / 2, 222 / 2 - 50, "Theseus's Escape", 0xFFFFFFFF).setXCentered(true).setVisibilitySupplier(() -> app.getGamestate() == 0));
+            app.addWidget(new ClickButtonWidget(323 / 2 - 10, 222 / 2 - 10, 30, 30, "Play",
+                    (clickData -> {
+                        app.setGamestate(1);
+                        app.resetGame();
+                    }))
+                    .setShouldClientCallback(true).setVisibilitySupplier(() -> app.getGamestate() == 0));
+            // Gamestate 1: Play
+            app.setMaze((MazeWidget) new MazeWidget().setVisibilitySupplier(() -> app.getGamestate() >= 1));
+            app.setPlayer((PlayerWidget) new PlayerWidget(0, 0, app).setVisibilitySupplier(() -> app.getGamestate() >= 1));
+            app.setEnemy((EnemyWidget) new EnemyWidget(-100, -100, app).setVisibilitySupplier(() -> app.getGamestate() >= 1));
+            // Gamestate 2: Pause
+            app.addWidget(new ImageWidget(5, 5, 333 - 10, 232 - 10, new ColorRectTexture(0xFF000000)).setVisibilitySupplier(() -> app.getGamestate() > 1));
+            app.addWidget(new ClickButtonWidget(323 / 2 - 10, 222 / 2 - 10, 50, 20, "Continue", (clickData) -> app.setGamestate(1)).setVisibilitySupplier(() -> app.getGamestate() == 2));
+            app.addWidget(new LabelWidget(333 / 2, 222 / 2 - 50, "Game Paused", 0xFFFFFFFF).setXCentered(true).setVisibilitySupplier(() -> app.getGamestate() == 2));
+            // Gamestate 3: Death
+            app.addWidget(new SimpleTextWidget(333 / 2, 232 / 2 - 40, "", 0xFFFFFFFF, () -> "Oh no! You were eaten by the Minotaur!", true).setVisibilitySupplier(() -> app.getGamestate() == 3));
+            app.addWidget(new SimpleTextWidget(333 / 2, 232 / 2 - 28, "", 0xFFFFFFFF, () -> "You got through " + app.getMazesSolved() + " mazes before losing.", true).setVisibilitySupplier(() -> app.getGamestate() == 3));
+            app.addWidget(new SimpleTextWidget(333 / 2, 232 / 2 - 16, "", 0xFFFFFFFF, () -> "Try again?", true).setVisibilitySupplier(() -> app.getGamestate() == 3));
+            app.addWidget(new ClickButtonWidget(323 / 2 - 10, 222 / 2 + 10, 40, 20, "Retry", (clickData -> {
+                app.setGamestate(1);
+                app.setMazesSolved(0);
+                MAZE_SIZE = 9;
+                speed = 25;
```

Comment:
Very good game interaction. Maybe you could also try `TerminalDialogWidget showInfoDialog (...). .setclientSide ().open()`. It is good for this kind of work.